src/Symfony/Component/Validator/ConstraintViolation.php

```diff
@@ -79,17 +79,30 @@ public function __toString()
         }
 
         $propertyPath = (string) $this->propertyPath;
-        $code = $this->code;
 
         if ('' !== $propertyPath && '[' !== $propertyPath[0] && '' !== $class) {
             $class .= '.';
         }
 
-        if (!empty($code)) {
-            $code = ' (code '.$code.')';
+        $context = '';
+        if (!empty($this->code)) {
+            $context .= ' code '.$this->code.', ';
+        }
+        $context .= 'invalid value ';
+        if (\is_object($this->invalidValue)) {
+            $invalidObjectClass = \get_class($this->invalidValue);
+            $context .= 'c' === $invalidObjectClass[0] && 0 === strpos($invalidObjectClass, "class@anonymous\0") ? get_parent_class($invalidObjectClass).'@anonymous' : $invalidObjectClass;
+        } elseif (\is_array($this->invalidValue)) {
+            $context .= 'array';
+        } elseif (\is_resource($this->invalidValue)) {
+            $context .= \get_resource_type($this->invalidValue);
+        } elseif (null === $this->invalidValue || \is_bool($this->invalidValue)) {
+            $context .= var_export($this->invalidValue, true);
+        } else {
+            $context .= (string) $this->invalidValue;
```

Comment:
when a string, should we really reflect its value? seems dangerous to me (like what when this string is 1GB long - or when it contains sensitive data, or is a hack payload, etc

Reply:
I understand your concern, since this method currently just dumps values which are under the validator responsibilities. Could we have at least an opt-in argument to choose if the invalid value should be dumped? My use case is based on the validation of a entity which has many constraints in their properties. So, in order to simplify the logging of failed validations, I'm logging something like this:
```php
implode("\n", iterator_to_array($violations));
```
Then, when get a violation in my log I know by instance, if I should update the validator to support an unhandled valid value (a false positive).
I know I could achieve this behavior by traversing the violations and calling `getInvalidValue()` by my own risk; but I thought it could be a nice addition for this method.